docs/interchange.rst

```diff
@@ -371,32 +440,50 @@ Valid tree sequence requirements
 ================================
 
 Arbitrary data can be stored in tables using the classes in the
-:ref:`sec_tables_api`. However, only tables that fulfil a set of
-requirements represent a valid :class:`.TreeSequence` object, and
-can be loaded using :func:`.load_tables`. In this
+:ref:`sec_tables_api`. However, only a :class:`.TableCollection`
+that fulfils a set of requirements represents
+a valid :class:`.TreeSequence` object which can be obtained
+using the :meth:`.TableCollection.tree_sequence` method. In this
 section we list these requirements, and explain their rationale.
 Violations of most of these requirements are detected when the
 user attempts to load a tree sequence via :func:`.load` or
-:func:`.load_tables`, raising an informative error message. Some
-more complex requirements may not be detectable at load-time,
+:meth:`.TableCollection.tree_sequence`, raising an informative
+error message. Some more complex requirements may not be detectable at load-time,
 and errors may not occur until certain operations are attempted.
 These are documented below.
 
+.. _sec_individual_requirements:
+
+Individual requirements
+-----------------------
+
+Individuals are a basic type in a tree sequence and are not defined with
+respect to any other tables. Therefore, there are no requirements on
+individuals.
+
+There are no requirements regarding the ordering of individuals.
+Sorting a set of tables using :meth:`.TableCollection.sort` has
+no effect on the individuals.
 
 .. _sec_node_requirements:
 
 Node requirements
 -----------------
 
-Nodes are the most basic type in a tree sequence, and are not defined with
-respect to any other tables. Therefore, the requirements for nodes are
-trivial.
+Given a valid set of individuals and populations, the requirements for
+each node are:
+
+- Node times must be non-negative;
+- ``population`` must either be null (-1) or refer to a valid population ID;
+- ``individual`` must either be null (-1) or refer to a valid individual ID.
+
+There are no requirements regarding the ordering of nodes with respect to time.
 
-- Node times must be non-negative.
+For simplicity and algorithmic efficiency, all nodes referring to the same
+(non-null) individual must be contiguous.
 
-There are no requirements regarding the ordering of nodes with respect to time
-or any other field. Sorting a set of tables using :func:`.sort_tables` has
-no effect on the nodes.
+Sorting a set of tables using :meth:`.TableCollection.sort`
+has no effect on nodes.
```

Comment:
I forget: what about adjacency of nodes in the same individual?  That's up to the user to guarentee? (which is ok for now...)

Reply:
Yes, that's up to the user for now. I think we should update TableCollection.sort() to sort nodes too so we can guarantee this, but for now it's fine.

(To address the concerns about extra time spent sorting nodes pointlessly, we can add a flag SKIP_NODES to sort for those who know their node tables are already OK.)